ffmpeg/ffmpeg.go

```diff
@@ -381,8 +381,11 @@ func (t *Transcoder) Transcode(input *TranscodeOptionsIn, ps []TranscodeOptions)
 func bypassTranscoding(input *TranscodeOptionsIn, ps []TranscodeOptions) (*TranscodeResults, error) {
 	glog.Warningf("Input %s has audio & video streams but no video frames. Bypassing the transcoder.", input.Fname)
 	for i, p := range ps {
-		if p.Profile != (VideoProfile{}) {
-			p.Profile = (VideoProfile{})
+		if p.Profile != (VideoProfile{}) && p.Profile.Format != FormatMP4 {
+			// we only bypass when a VideoProfile is set (i.e. not a copy/drop) and output format
+			// is mpegts.
+			// FIXME: Temporary fix. The transcoder is not the place for such hacky code
+			p.Profile = VideoProfile{Name: p.Profile.Name, Format: FormatMPEGTS}
```

Comment:
We'll skip the non-encoding related operations such as the scale & FPS filter if we initialize a new video profile here and only set the name/format. Is this desired behavior?

My initial thought is that we would want to still apply filters configured in the video profile fields, but just use the copy video & audio encoder. Or is it the case that the filters would not work properly for the first audio-only segment in a session so the only thing we can do is to pass the segment to the copy video & audio encoder?

Reply:
> or is it the case that the filters would not work properly for the first audio-only segment in a session so the only thing we can do is to pass the segment to the copy video & audio encoder?

Yeah exactly! Filters get applied on decoded frames, and there are no video frames here for the audio-only segments :)
Although the subsequent normal segments will not go through this bypass function, and would get filtered/transcoded as usual.

There's no way to even set the filters on in the `copy` mode https://stackoverflow.com/questions/53518589/how-to-use-filtering-and-stream-copy-together-with-ffmpeg

We can try enabling the audio-filters here (which don't really do much) but I wanna avoid that at first and only consider it if we face issues, because doing that will likely complicate the delicate re-muxing further.